### PR TITLE
fix(getThreadException): Fix edge case issue

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes: https://app.asana.com/0/1158284503473033/1198985561863021

**Problem:** 
If a user sets an Exception and a Thread(s), but the exception has no stack trace (the threads do though), no stack trace is shown.

**Solution:**
if the exception has no stack trace, the `getThreadException` function will return `undefined`. If `getThreadException` used inside the function `getThreadStacktrace`  returns `undefined`, the function `getThreadStacktrace` will return the stack trace or the raw stack trace, if one of the two is defined.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/crashContent/index.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ type Props = Pick<
   'stackType' | 'stackView' | 'projectId' | 'event' | 'newestFirst'
 > & {
   exception?: ExceptionType;
-  stacktrace?: React.ComponentProps<typeof Stacktrace>['stacktrace'];
+  stacktrace?: ExceptionValue['stacktrace'];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/interfaces/crashContent/index.tsx:14]: these changes were necessary in this PR. I also intend to clean these types in a follow-up PR

File: src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadException.tsx [hunk 2]
```diff
@@ -2,27 +2,33 @@ import {Thread} from 'app/types/events';
...
   if (!exceptionDataValues?.length) {
     return undefined;
   }
 
   if (exceptionDataValues.length === 1 && !defined(exceptionDataValues[0].threadId)) {
-    return exceptionData;
+    if (!exceptionDataValues[0].stacktrace) {
+      return undefined;
+    }
+    return exceptionData as Required<ExceptionType>;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadException.tsx:26]: Should there be tests for this change in logic?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadException.tsx:26]: sure thing!

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/getThreadException.tsx:26]: done :)

**Review by markstory — APPROVED** [review]

**Pull request merged**